tests/drivers/gpio/gpio_get_direction/src/main.c

```diff
@@ -6,108 +6,117 @@
 
 #include "test_gpio_get_direction.h"
 
-static void test_disconnect(void)
+struct gpio_get_direction_fixture {
+	const struct device *port;
+	gpio_pin_t pin;
+	gpio_flags_t flags;
+};
+
+static void *gpio_get_direction_setup(void)
+{
+	static struct gpio_get_direction_fixture fixture;
+
+	fixture.pin = TEST_PIN;
+
+	return &fixture;
+}
+
+static void gpio_get_direction_before(void *arg)
+{
+	struct gpio_get_direction_fixture *fixture = (struct gpio_get_direction_fixture *)arg;
+
+	fixture->port = device_get_binding(TEST_DEV);
```

Comment:
No need to make a change here, but generally speaking the call to `device_get_binding` can be made in the setup since you only need it once per suite, then the assert can be in the before since it can't go in the setup.

Reply:
Fixed!